apis/v1alpha1/gatewayclass_types.go

```diff
@@ -114,6 +114,10 @@ const (
 	// validity of the Parameters set for a given controller. This
 	// will initially start off as "Unknown".
 	GatewayClassConditionStatusInvalidParameters GatewayClassConditionType = "InvalidParameters"
+
+	// FinalizerGatewaysExist should be added as a finalizer to the GatewayClass
+	// whenever there are provisioned Gateways using a GatewayClass.
+	FinalizerGatewaysExist = "Gateways using GatewayClass exist"
```

Comment:
Nit: To match the var name above should this be `GatewayClassFinalizerGatewaysExist`? I know that is pretty long though...

Reply:
The example in the [k8s docs](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#finalizers) seems to suggest that a finalizer name should be more structured. Maybe something like `gateway-finalizer.networking.x-k8s.io`?

Reply:
> GatewayClassFinalizerGatewaysExist

The reason I chose FinalizerName was to catagorize all the finalizers together (makes lookup easier). Granted that we only have a single finalizer in this package currently.

> The example in the k8s docs seems to suggest that a finalizer name should be more structured. Maybe something like gateway-finalizer.networking.x-k8s.io?

Updated.

Reply:
Wouldn't finalizers be applied to a specific resource though? Seems like they could be used similarly to conditions which are also resource specific. If finalizers were globally applicable it might make more sense to specify them in a separate file.

Reply:
I see. That is a good point. I've updated this to be `GatewayClassFinalizerGatewaysExist`. Not the best name but considering that this won't be used very often, it should be okay? I'm open to better names.